File: Algorithm_python/Baekjun/Category/BinarySearch/1920.py
# 4 1 5 2 3 <- arr
# 1 2 3 4 5 <- arr.sort()
# 0 1 2 3 4 <- index

# 1 3 7 9 5
def binary_search(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1

N = int(input())
arr = list(map(int, input().split()))
M = int(input())
targets = list(map(int, input().split()))

arr.sort()
print(arr)

for target in targets:
    if binary_search(arr, target) != -1:
        print(1)
    else:
        print(0)

File: Algorithm_cpp/Baekjun/Category/BinarySearch/1920/1920.cpp
#include <iostream>
#include <algorithm>
using namespace std;

void binary_search(int arr[], int n, int target)
{
    bool res = false;
    int left = 0;
    int right = n - 1;

    while( left <= right){
        int mid = (left + right) / 2;
        if (arr[mid] == target) {
            res = true;
        }
        else if (arr[mid] < target){
            left = mid + 1;
        }
        else {
            right = mid - 1;
        }

    }

    if (res) {
        cout << "1\n";
    } else {
        cout << "0\n";

}


void main(int argc, char* argv[])
{
    int n = 0;
    cin >> n;

    int arr[n];
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    sort(arr, arr + n);

    int m = 0;
    cin >> m;

    for (int i = 0; i < m; i++) {
        int target = 0;
        cin >> target;

        // Binary search
        if (binary_search(arr, arr + n, target)) {
            cout << "1\n";
        } else {
            cout << "0\n";
        }
    }
}